src/o365/spo/commands/theme/theme-list.ts

```diff
@@ -0,0 +1,118 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as request from 'request-promise-native';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: GlobalOptions;
+}
+
+class ThemeListCommand extends SpoCommand {
+
+  public get name(): string {
+    return commands.THEME_LIST;
+  }
+
+  public get description(): string {
+    return 'Retreives the list of custom themes added to the tenant store.';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}. Loading themes from tenant store...`);
+        }
+
+        if (this.verbose) {
+          cmd.log(`Retrieving themes from tenant store...`);
+        }
+
+        const requestOptions: any = {
+          url: `${auth.site.url}/_api/thememanager/GetTenantThemingOptions`,
+          method: 'POST',
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${accessToken}`,
+            'accept': 'application/json;odata=nometadata'
+          }),
+          body: '',
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+        return request.get(requestOptions);
+      })
+      .then((rawRes: any): void => {
+
+        if (args.options.debug) {
+          cmd.log('Response:');
+          cmd.log(rawRes);
+          cmd.log('');
+        }
+
+        const themePreviews: any[] = rawRes.themePreviews;
+
+          if (themePreviews && themePreviews.length > 0) {
+
+            if (args.options.output === 'json') {
+              cmd.log(themePreviews);
+            }
+            else {
+              try {
+                themePreviews.map(a => {
+                  const themeJson = JSON.parse(a.themeJson);
+                  cmd.log(`Name: ${a.name}\nPalette: ${JSON.stringify(themeJson.palette)}\nInverted: ${(themeJson.isInverted).toString()}\n`);              
```

Comment:
When printing multiple objects in the CLI, we don't format them manually into a string. Instead, we pass the array of objects to the `cmd.log` method which then uses consistent formatting across the CLI. I'd say that for the overview, let's print only the names of the available themes. Additional information like the palette or inverted can be retrieved using the `spo theme get` command.

Reply:
Ok, will alter print and will add new command `spo theme get`